sale_elaboration/tests/test_sale_elaboration.py

```diff
@@ -11,9 +11,25 @@ class TestSaleElaboration(AccountTestInvoicingCommon):
     def setUpClass(cls):
         super().setUpClass()
         cls.Elaboration = cls.env["product.elaboration"]
+        cls.ElaborationProfile = cls.env["product.elaboration.profile"]
+        cls.parent_category = cls.env["product.category"].create(
+            {"name": "Test Parent Category"}
+        )
+        cls.child_category = cls.env["product.category"].create(
+            {"name": "Test Parent Category", "parent_id": cls.parent_category.id}
+        )
         cls.product = cls.env["product.product"].create(
             {"name": "test", "tracking": "none", "list_price": 1000}
         )
+        cls.product_2 = cls.env["product.product"].create(
+            {"name": "test 2", "tracking": "none", "list_price": 1000}
+        )
+        cls.product_3 = cls.env["product.product"].create(
+            {"name": "test 2", "tracking": "none", "list_price": 1000}
+        )
+        cls.product.categ_id = cls.child_category
+        cls.product_2.categ_id = cls.child_category
+        cls.product_3.categ_id = cls.parent_category
```

Comment:
I think you can play with `cls.product_elaboration_A`  and `cls.product_elaboration_B` so you don't need to declare that many records.

Then, in your test cases, you can assign them to the categories to avoid modifiying the existing cases behavior.